lib/bsdlib/nrf91_sockets.c

```diff
@@ -406,6 +406,8 @@ static int nrf_to_z_dns_error_code(int nrf_error)
 		return DNS_EAI_NONAME;
 	case NRF_EINPROGRESS:
 		return DNS_EAI_INPROGRESS;
+	case NRF_ENETUNREACH:
+		return DNS_EAI_SERVICE;
```

Comment:
Is this a good error code here? `DNS_EAI_SERVICE` is directly related to the `service` argument of the `getaddrinfo` call (and indicates it was not a valid one). IMO `ENETUNREACH` falls under `DNS_EAI_SYSTEM` (we should set the `errno` variable in that case though, which from what I see is not done).

Reply:
Yes, I agree, I misset the part in the getaddrinfo documentation that it can return errnos. Fixing